scripts/jenkins/cloud/ansible/roles/list_packages/tasks/list-packages.yml

```diff
@@ -29,12 +29,21 @@
     loop_var: item_virtual_hosts
   register: _result_packages_rpm
 
+# delegate_facts used to get hostvars from delegate hosts
 - name: Collect list of installed packages repo of origin from zypper
   shell: |
-    zypper --xmlout se -s -i -t package
+    if [ {{ hostvars[item_virtual_hosts].ansible_os_family }} = "Suse" ];then
+      zypper --xmlout se -s -i -t package
+    else
+      echo "<?xml version='1.0'?>"
+      echo "<stream>"
+      repoquery -a --installed --qf='<solvable name="%{NAME}" version="%{VERSION}" release="%{RELEASE}" repository="%{REPO}"/>'
+      echo "</stream>"
+    fi
   args:
     warn: False
   delegate_to: "{{ item_virtual_hosts }}"
+  delegate_facts: True
```

Comment:
Shouldn't the `delegate_facts` be on a task that is gathering the facts together with a `delegate_to` for the virtual hosts? Where are you gathering the facts for the virtual hosts?

Reply:
I m not sure what do u mean. There is no task specifically for gathering facts from virtual host in this role to keep it minimalistic(e.g. where [needed](https://docs.ansible.com/ansible/latest/user_guide/playbooks_delegation.html#delegated-facts)). There is only `gather_facts`  in the playbook `list-or-diff-installed-packages.yml` . Do u want it to be role or plabook wide? I m not sure if we always want to delegate facts.

Reply:
The facts were gather already and are in `hostvars[item_virtual_hosts]`, no? I agree with @flaviodsr that `delegate_facts` doesn't do anything in this task.

Reply:
@nicolasbock yup. When I tested it before it failed to get the correct facts but now with different setup it works so I removed `delegate_facts` from here.